### ISSUE TITLE
[SfDataGridView] incorrect dispose implementation in scrollview_widget.dart

### ISSUE THREAD

**Issue by alexgarbarev** [syncfusion/flutter-widgets#1327]
Hello,
I'd like to raise an issue with current `dispose()` implementation inside `scrollview_widget.dart`:

```dart
  @override
  void dispose() {
    if (_verticalController != null &&
        _dataGridConfiguration.disposeVerticalScrollController) {
      _verticalController!
        ..removeListener(_verticalListener)
        ..dispose();
    }

    if (_horizontalController != null &&
        _dataGridConfiguration.disposeHorizontalScrollController) {
      _horizontalController!
        ..removeListener(_horizontalListener)
        ..dispose();
    }

    super.dispose();
  }
 ```
 
 As you can see, `removeListener` is not called, when `disposeVerticalScrollController` flag is false (when user provides own scroll controller). 
This leads to memory leak and crashes in console (because listener is not removed, it tries to call `setState` on disposed widget)
 
 Correct implementation should be:
 ```dart
  @override
  void dispose() {
    _verticalController?.removeListener(_verticalListener);
    _horizontalController?.removeListener(_horizontalListener);

    if (_verticalController != null &&
        _dataGridConfiguration.disposeVerticalScrollController) {
      _verticalController!.dispose();
    }

    if (_horizontalController != null &&
        _dataGridConfiguration.disposeHorizontalScrollController) {
      _horizontalController!.dispose();
    }

    super.dispose();
  }
  ```

So:
* we should only call `dispose` when we are the owner of scrollController (when we are creator)
* we should always call `removeListener`, to avoid callbacks on disposed widgets.

**Comment by Tamilarasan-Paranthaman** [comment]
Hi @alexgarbarev,

Thank you for bringing this matter to our attention. We are pleased to inform you that we have promptly resolved this matter in our most recent DataGrid version. To ensure you benefit from this fix, we kindly request that you update the [syncfusion_flutter_datagrid](https://pub.dev/packages/syncfusion_flutter_datagrid) package to version **22.2.9**.

Regards,
Tamilarasan

**Comment by ashok-kuvaraja** [comment]
Hi @alexgarbarev,

We suspect that the reported problem has been resolved in the DataGird package version 22.2.9. We are closing this issue. Please let us know if you have any further queries on this. We are happy to help you.

Regards, 
Ashok K
